src/Client/index.ts

```diff
@@ -124,6 +127,28 @@ export class Client {
       insecure: options.tls === false,
     };
 
+    if (options.tlsCAFile) {
+      if (channelCredentials.insecure) {
+        debug.connection(
+          "tslCAFile passed to insecure connection. Will be ignored."
+        );
+      } else {
+        const resolvedPath = isAbsolute(options.tlsCAFile)
+          ? options.tlsCAFile
+          : resolve(process.cwd(), options.tlsCAFile);
+
+        debug.connection(`Resolved tslCAFile option as ${resolvedPath}`);
+
+        if (!existsSync(resolvedPath)) {
+          throw new Error(
+            "Failed to load certificate file. File was not found."
+          );
+        }
+
+        channelCredentials.rootCertificate = readFileSync(resolvedPath);
+      }
```

Comment:
**SIDENOTE:** We're missing the last point from RFC:

> 4. If a file cannot be read because of the wrong format, error with message Failed to read certificate file. File is in the wrong format. should be thrown/returned.